manifests/server.pp

```diff
@@ -27,35 +27,35 @@
 #  }
 #
 class puppet::server (
-  $autosign           = undef,
-  $bindaddress        = '0.0.0.0',
-  $ca                 = false,
-  $config_version_cmd = '/usr/bin/git --git-dir $confdir/environments/$environment/.git rev-parse --short HEAD 2>/dev/null || echo',
-  $dns_alt_names      = undef,
-  $enc                = '',
-  $enc_exec           = '',
-  $ensure             = 'present',
-  $directoryenvs      = true,
-  $environmentpath    = undef,
-  $basemodulepath     = [],
-  $default_manifest   = undef,
-  $manage_package     = true,
-  $manifest           = '$confdir/modules/site/site.pp',
-  $modulepath         = ['$confdir/modules/site', '$confdir/env/$environment/dist'],
-  $parser             = undef,
-  $manage_puppetdb    = false,
-  $report             = true,
-  $report_dir         = $puppet::params::report_dir,
-  $reportfrom         = undef,
-  $reports            = ['store', 'https'],
-  $reporturl          = "https://${::fqdn}/reports",
-  $servername         = $::fqdn,
-  $serverssl_ciphers  = undef,
-  $serverssl_protos   = undef,
-  $servertype         = 'unicorn',
-  $storeconfigs       = undef,
-  $stringify_facts    = false,
-  $package            = $puppet::params::master_package,
+  $autosign          = undef,
+  $bindaddress       = '0.0.0.0',
+  $ca                = false,
+  $config_version    = '/usr/bin/git --git-dir $confdir/environments/$environment/.git rev-parse --short HEAD 2>/dev/null || echo',
```

Comment:
Yes, when `puppet::server::directoryenvs` is true, then the `manifest`, `modulepath` and `config_version` variables get removed from puppet.conf (lines 32-34 in `puppet::server::config` class)